### ISSUE TITLE
Additional fixed checks for the translation terms

### ISSUE THREAD

**Issue by mkaring** [tom-englert/ResXResourceManager#236]
Hello,

I'd like to suggest adding some additional reasonable validations for the translated terms, such as the validation of the format parameters.

The following checks I would propose to add:

- Check if the white-spaces at the start and the end of the string match the input. Meaning if the neutral language term starts with a line break, the translation should too.
  The use-case for me is that in the project I am working I noticed some translations, where the translations have spaces and line breaks at the end, but the neutral language term does not. This is a translation error, but on that is not easily spotted.
- Similar to the white-spaces: Check the punctuation at the start and the end of the term. If the neutral language term ends on a colon, the translation should too.

These two checks are very general and I think can be used without many issues for all other users.

Additional checks, such as verifying that the length of the neutral term and the translation doesn't differ too much or that the quotation in the terms match may lead to false positives. Those shouldn't be inserted, unless there is a way to disable specific checks.

For the two checks above I volunteer to implement them and send a PR your way, provided you agree with their usefulness.

**Comment by tom-englert** [comment]
Sounds reasonable, if they are both optional (opt in via configuration tab)

**Comment by mkaring** [comment]
I'm on it.

**Comment by tom-englert** [comment]
Are you sure this will work for all languages? 

I doubt that Chinese always use the same spacing and punctuation than e.g. German or English.

However if it's optional, you can switch it on only if it works for all of your used languages.

**Comment by mkaring** [comment]
I noticed that already. I added some "normalization" for the punctuation checks so it works okay for languages like Chinese. These checks will require some tweaking once user feedback comes in.

My plan is currently to have a global configuration for each rule so it can be enabled and disabled and to allow the rules to be muted using a `@MuteRule(...)` entry in the comment field, similar to `@Invariant`.
